processor/signozlogspipelineprocessor/processor.go

```diff
@@ -0,0 +1,209 @@
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/extension/experimental/storage"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+
+	_ "github.com/SigNoz/signoz-otel-collector/pkg/parser/grok" // ensure grok parser gets registered.
+)
+
+func newProcessor(
+	processorConfig *Config,
+	nextConsumer consumer.Logs,
+	telemetrySettings component.TelemetrySettings,
+) (*logsPipelineProcessor, error) {
+
+	sink := &stanzaToOtelConsumer{
+		nextConsumer: nextConsumer,
+		logger:       telemetrySettings.Logger,
+	}
+
+	stanzaPipeline, err := pipeline.Config{
+		Operators:     processorConfig.BaseConfig.Operators,
+		DefaultOutput: sink,
+	}.Build(telemetrySettings)
+	if err != nil {
+		return nil, err
+	}
+
+	return &logsPipelineProcessor{
+		telemetrySettings: telemetrySettings,
+		nextConsumer:      nextConsumer,
+
+		processorConfig: processorConfig,
+		stanzaPipeline:  stanzaPipeline,
+
+		sink: sink,
+	}, nil
+}
+
+// Implements processor.Logs
+type logsPipelineProcessor struct {
+	telemetrySettings component.TelemetrySettings
+	nextConsumer      consumer.Logs
+
+	processorConfig *Config
+	stanzaPipeline  *pipeline.DirectedPipeline
+	firstOp         operator.Operator
+
+	shutdownFns []component.ShutdownFunc
+
+	sink *stanzaToOtelConsumer
+}
+
+// Collector starting up
+func (p *logsPipelineProcessor) Start(ctx context.Context, _ component.Host) error {
+	p.telemetrySettings.Logger.Info("Starting logsPipelineProcessor")
+
+	err := p.stanzaPipeline.Start(storage.NewNopClient())
+	if err != nil {
+		return err
+	}
+
+	p.shutdownFns = append(p.shutdownFns, func(_ context.Context) error {
+		return p.stanzaPipeline.Stop()
+	})
+
+	// .Operators() returns topologically sorted operators for the stanza operator graph
+	// logs will be fed into p.firstOp for processing
+	p.firstOp = p.stanzaPipeline.Operators()[0]
+
+	return nil
+}
+
+// Collector shutting down
+func (p *logsPipelineProcessor) Shutdown(ctx context.Context) error {
+	p.telemetrySettings.Logger.Info("Stopping logsPipelineProcessor")
+
+	// Call shutdown functions in reverse order, so that shutdown happens in LIFO order of starting
+	for i := len(p.shutdownFns) - 1; i >= 0; i-- {
+		fn := p.shutdownFns[i]
+
+		if err := fn(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// baseConsumer interface
+func (p *logsPipelineProcessor) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: true}
+}
+
+func (p *logsPipelineProcessor) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	p.telemetrySettings.Logger.Debug(
+		"logsPipelineProcessor received logs",
+		zap.Any("logs", ld),
+	)
+
+	entries := plogToEntries(ld)
+
+	for _, e := range entries {
+		if err := p.firstOp.Process(ctx, e); err != nil {
+			p.telemetrySettings.Logger.Error("processor encountered an issue with the pipeline", zap.Error(err))
+		}
+	}
+
+	// Flush processed entries as a single plog.Logs to next consumer.
+	return p.sink.flush(ctx)
```

Comment:
Should we return an error where there is an error with `p.firstOp.Process(ctx, e)` than reaching here and flushing?